Arm64 tests are failing to run on the correct arch

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Since Jan 13th, the [daily CI](https://github.com/rook/rook/actions/workflows/daily-nightly-jobs.yml) has been failing the arm64 tests. Before that, they were passing.

This appears related to the merge of #16886 when the minikube config changed.

**Expected behavior:**
Green CI

**How to reproduce it (minimal and precise):**
<!-- Please let us know any circumstances for reproduction of your bug. -->
See the [daily CI](https://github.com/rook/rook/actions/workflows/daily-nightly-jobs.yml) history.

```
Run curl -LO https://github.com/Mirantis/cri-dockerd/releases/download/v0.3.21/cri-dockerd_0.3.21.3-0.ubuntu-focal_amd64.deb
+ curl -LO https://github.com/Mirantis/cri-dockerd/releases/download/v0.3.21/cri-dockerd_0.3.21.3-0.ubuntu-focal_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

 31 10.2M   31 3294k    0     0  4554k      0  0:00:02 --:--:--  0:00:02 4554k
100 10.2M  100 10.2M    0     0  10.2M      0 --:--:-- --:--:-- --:--:-- 25.5M
+ sudo dpkg -i cri-dockerd_0.3.21.3-0.ubuntu-focal_amd64.deb
dpkg: error processing archive cri-dockerd_0.3.21.3-0.ubuntu-focal_amd64.deb (--install):
 package architecture (amd64) does not match system (arm64)
Errors were encountered while processing:
 cri-dockerd_0.3.21.3-0.ubuntu-focal_amd64.deb
Error: Process completed with exit code 1.
```

--- a/.github/workflows/integration-test-setup-cluster-resources/action.yaml
+++ b/.github/workflows/integration-test-setup-cluster-resources/action.yaml
@@ -25,9 +25,10 @@ runs:
     - name: Install cri-dockerd
       shell: bash --noprofile --norc -eo pipefail -x {0}
       run: |
-        curl -LO https://github.com/Mirantis/cri-dockerd/releases/download/v0.3.21/cri-dockerd_0.3.21.3-0.ubuntu-focal_amd64.deb
-        sudo dpkg -i cri-dockerd_0.3.21.3-0.ubuntu-focal_amd64.deb
-        rm -f cri-dockerd_0.3.21.3-0.ubuntu-focal_amd64.deb
+        ARCH=$(go env GOARCH)
+        curl -LO https://github.com/Mirantis/cri-dockerd/releases/download/v0.3.21/cri-dockerd_0.3.21.3-0.ubuntu-focal_${ARCH}.deb
+        sudo dpkg -i cri-dockerd_0.3.21.3-0.ubuntu-focal_${ARCH}.deb
+        rm -f cri-dockerd_0.3.21.3-0.ubuntu-focal_${ARCH}.deb
 
     - name: Setup Minikube
       uses: medyagh/setup-minikube@latest

